File: summer/WebContent/js/system/windows/ResourceKey.js
/**
 * ResourceKey
 */

define(["dojo/_base/declare", "system/Type", "markup/MarkupExtension"], 
		function(declare, Type, MarkupExtension){
	var ResourceKey = declare("ResourceKey", MarkupExtension,{
		constructor:function(){
		},
		
		/// <summary> 
        ///  Return this object.  ResourceKeys are typically used as a key in a dictionary.
        /// </summary> 
//        public override object 
		ProvideValue:function(/*IServiceProvider*/ serviceProvider) 
        {
            return this; 
        }
	});
	
	Object.defineProperties(ResourceKey.prototype,{
        /// <summary>
        ///     Used to determine where to look for the resource dictionary that holds this resource. 
        /// </summary> 
//        public abstract Assembly 
		Assembly:
        { 
            get:function(){}
        }
	});
	
	ResourceKey.Type = new Type("ResourceKey", ResourceKey, [MarkupExtension.Type]);
	return ResourceKey;
});

File: org.summer.view.widget/src/org/summer/view/widget/ResourceKey.java
package org.summer.view.widget;

import org.summer.view.widget.markup.MarkupExtension;

/// <summary> 
///     Abstract base class for various resource keys.
///     Provides a common base for simple key detection in resource components. 
/// </summary>
//[MarkupExtensionReturnType(typeof(ResourceKey))]
public abstract class ResourceKey extends MarkupExtension
{ 
    /// <summary>
    ///     Used to determine where to look for the resource dictionary that holds this resource. 
    /// </summary> 
    public abstract Assembly Assembly
    { 
        get;
    }

    /// <summary> 
    ///  Return this object.  ResourceKeys are typically used as a key in a dictionary.
    /// </summary> 
    public /*override*/ Object ProvideValue(IServiceProvider serviceProvider) 
    {
        return this; 
    }
}